tensorboard/webapp/runs/store/runs_reducers_test.ts

```diff
@@ -238,7 +247,67 @@ describe('runs_reducers', () => {
           ['lambda', colorUtils.CHART_COLOR_PALLETE[3]],
         ])
       );
-      expect(nextState.data.nextGroupColorIndex).toBe(4);
+    });
+
+    describe('advanced grouping', () => {
+      it('assigns default color to by experiment', () => {
+        const state = buildRunsState({
+          groupBy: {
+            key: GroupByKey.EXPERIMENT,
+          },
+          defaultRunColor: new Map([
+            ['foo', '#aaa'],
+            ['bar', '#aaa'],
+          ]),
+          groupToColor: new Map([['eid1', '#aaa']]),
+        });
+        const action = actions.fetchRunsSucceeded({
+          experimentIds: ['eid1', 'eid2'],
+          runsForAllExperiments: [
+            buildRun({id: 'baz'}),
+            buildRun({id: 'foo'}),
+            buildRun({id: 'qaz'}),
+            buildRun({id: 'alpha'}),
+            buildRun({id: 'beta'}),
+            buildRun({id: 'gamma'}),
+            buildRun({id: 'lambda'}),
+          ],
+          newRunsAndMetadata: {
+            eid1: {
+              runs: [
+                buildRun({id: 'baz'}),
+                buildRun({id: 'foo'}),
+                buildRun({id: 'qaz'}),
```

Comment:
'foo' is not in this experiment but 'foo' is assigned the same color? Or the default status shows the 'foo' is also under the same eid?

Reply:
This is testing the case when a `run` is already present in the store with its value assigned.